src/main/java/io/molr/gui/fx/widgets/MissionInstancesView.java

```diff
@@ -82,18 +86,51 @@ public void init() {
             mole.instruct(selectedInstance.handle(), MissionCommand.DISPOSE);
         });
         buttons.getChildren().add(disposeButton.getButton());
-
-        listView.getSelectionModel().getSelectedItems().addListener((ListChangeListener<MissionInstance>) c -> {
-            disposeButton.getButton().setDisable(true);
-            if (c.getList().size() == 1) {
-                MissionInstance selectedInstance = c.getList().get(0);
-                onSelectedMissionInstanceChange(selectedInstance);
+        
+        listView.getSelectionModel().getSelectedItems().addListener(new ListChangeListener<MissionInstance>() {
+            @Override
+            public void onChanged(Change<? extends MissionInstance> c) {
+                disposeButton.getButton().setDisable(true);
+                if(c.getList().size()==1) {
+                    MissionInstance selectedInstance = c.getList().get(0);
+                    onSelectedMissionInstanceChange(selectedInstance);
+                }
             }
         });
 
         setBottom(buttons);
     }
-
+    
+    private void onStatesUpdate(AgencyState agencyState) {
+        activeMissions.setAll(agencyState.activeMissions());
+    }
+    
+    private void subscribeToMoleStates() {
+        LOGGER.info("subscribe to mole states");
+        mole.states().publishOn(FxThreadScheduler.instance()).subscribe(this::onStatesUpdate, this::onStatesError, this::onStatesComplete);
+    }
+    
+    private void onStatesError(Throwable throwable) {
+        LOGGER.info("onStatesErrorMissionInstances\n"+throwable);
```

Comment:
This looks like duplicated code as in Available missions view.... Can we do this in the client class? And even extract a utility method to do this?